File: evidence/generated-systems/golang/insurance-on-golang/insurance-on-golang/internal/model/PolicyCoverage.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// PolicyCoverage Declaration
//==============================================================
type PolicyCoverage struct {
    gorm.Model
     Limit                                                            string
    Deductible                                                            string
    Premium                                                            string
    PolicyId         *uint
    Policy           *Policy `gorm:"foreignKey:PolicyId"`
     InsuredObjects           []InsuredObject `gorm:"foreignKey:InsuredObjectsFromPolicyCoverageId"`
    CoverageType                      CoverageType

// parent associations as their child

}

File: evidence/generated-systems/django/insurance-on-django/insuranceOnDjango/models/PolicyCoverage.py
from django.db import models
from insuranceOnDjango.models.CoverageType import CoverageType

#======================================================================
# 
# Encapsulates data for model PolicyCoverage
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class PolicyCoverage Declaration
#======================================================================
class PolicyCoverage (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	limit = Money
	deductible = Money
	premium = Money
	policy = models.ForeignKey('Policy', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	insuredObjects = models.ManyToManyField('InsuredObject',  blank=True, related_name='+')
	coverageType = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in CoverageType])

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.limit
		str = str + self.deductible
		str = str + self.premium
		str = str + self.coverageType
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "PolicyCoverage";
    
	def objectType(self):
		return "PolicyCoverage";
